scripts/build.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+yarn tsc --project tsconfig.build.json
+mkdir tmp
+touch tmp/cli.js
+echo "#!/usr/bin/env node" >> tmp/cli.js
```

Comment:
Rather than dynamically add the shebang, can we add it directly to `cli.js`?

Reply:
I thought about adding it to `cli.ts`, but I thought that would be confusing, because that file can't be run through Node, it has to be run through `ts-node` instead.

Reply:
Ah I see. Makes sense. Hmm. Maybe we could use a wrapper Node.js script instead? e.g.

```javascript
#!/usr/bin/env node

require('./dist/cli.js')()
```

That would also remove the need for a build script.

Reply:
Okay, do you propose still building this on the fly in `scripts/build.sh` or putting this in `src/`?

Reply:
Oh I guess it could go in `src`. Another pattern I've seen before is creating a `bin` directory for the wrapper script. Either way, it avoids the need to move anything around at build time.

Reply:
I've added a script to `bin/`.